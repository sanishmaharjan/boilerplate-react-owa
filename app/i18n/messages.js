export default {
    en: {
        LOGOUT: 'Logout',
        CHOOSE_LANGUAGE: 'Choose language',
        SUCCESS: 'Success',
        ERROR: 'Error',
        INFO: 'Info',
        HOME_PAGE: 'Home page',
        BLINK_PAGE: 'Blink page',
        USER_INFORMATION: 'User Information',
        USER_UUID: 'User Uuid',
        NAME: 'Name',
        USER_NAME: 'User Name',
        NOTIFICATION: 'Notification',
        MESSAGE: 'Message'
    },
    es: {
        LOGOUT: 'Cerrar sesión',
        CHOOSE_LANGUAGE: 'Elige lengua',
        SUCCESS: 'Éxito',
        ERROR: 'Error',
        INFO: 'Información',
        HOME_PAGE: 'Página de inicio',
        BLINK_PAGE: 'Página de parpadeo',
        USER_INFORMATION: 'informacion del usuario',
        USER_UUID: 'Usuario Uuid',
        NAME: 'Nombre',
        USER_NAME: 'Nombre de usuario',
        NOTIFICATION: 'Notificación',
        MESSAGE: 'Mensaje'
    },
    fr: {
        LOGOUT: 'Connectez - Out',
        CHOOSE_LANGUAGE: 'Choisissez la langue',
        SUCCESS: 'Succès',
        ERROR: 'Erreur',
        INFO: 'Info',
        HOME_PAGE: 'Page d\'accueil',
        BLINK_PAGE: 'Blink page',
        USER_INFORMATION: 'informations de l\'utilisateur',
        USER_UUID: 'Utilisateur Uuid',
        NAME: 'prénom',
        USER_NAME: 'Nom d\'utilisateur',
        NOTIFICATION: 'Notification',
        MESSAGE: 'Message'
    },
    pt: {
        LOGOUT: 'Sair',
        CHOOSE_LANGUAGE: 'Escolha o seu idioma',
        SUCCESS: 'Sucesso',
        ERROR: 'Erro',
        INFO: 'Info',
        HOME_PAGE: 'Pagina inicial',
        BLINK_PAGE: 'Página de piscar',
        USER_INFORMATION: 'Informação do usuário',
        USER_UUID: 'Usuário Uuid',
        NAME: 'Nome',
        USER_NAME: 'Nome de usuário',
        NOTIFICATION: 'Notificação',
        MESSAGE: 'mensagem'
    }
};
