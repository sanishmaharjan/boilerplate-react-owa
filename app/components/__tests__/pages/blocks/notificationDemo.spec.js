import React from 'react';
import {shallow, mount} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import {IntlProvider} from 'react-intl';

import NotificationDemo from 'components/pages/blocks/notificationDemo';
import messages from 'i18n/messages';

const intlProvider = new IntlProvider({locale: 'en', messages: messages['en']}, {});
const {intl} = intlProvider.getChildContext();
const testData = {
    context: {
        intl: intl,
        store: {
            notify: (type, message) => jest.fn(),
        }
    },
    sleep: (milisec) => {
        return new Promise((resolve, reject) => {
            setTimeout(() => {
                return resolve(true);
            }, milisec);
        });
    }
};

describe('NotificationDemo', () => {
    it('renders correctly', () => {
        const notificationDemo = shallow(<NotificationDemo defaultMsg="This is sample message" />, {context: testData.context});

        expect(shallowToJson(notificationDemo)).toMatchSnapshot();

        notificationDemo.setState({message: 'New message'});
        testData.sleep(100);
        expect(notificationDemo.find('input').props().value).toBe('New message');
        expect(shallowToJson(notificationDemo)).toMatchSnapshot();
    });

    it('Should trigger event handler', async () => {
        const spyOnChangeMsgField = jest.spyOn(NotificationDemo.prototype, 'onChangeMsgField');
        const spyOnSuccessMsg = jest.spyOn(NotificationDemo.prototype, 'successMsg');
        const spyOnErrorMsg = jest.spyOn(NotificationDemo.prototype, 'errorMsg');
        const spyOnInfoMsg = jest.spyOn(NotificationDemo.prototype, 'infoMsg');
        const spySetState = jest.spyOn(NotificationDemo.prototype, 'setState');
        const spyNotify = jest.spyOn(testData.context.store, 'notify');
        const notificationDemo = mount(<NotificationDemo defaultMsg="This is sample message" />, {context: testData.context});

        notificationDemo.find('#success-btn').simulate('click');
        expect(spyOnSuccessMsg).toHaveBeenCalled();
        expect(spyNotify).toHaveBeenCalledWith('success', 'This is sample message');

        notificationDemo.find('#error-btn').simulate('click');
        expect(spyOnErrorMsg).toHaveBeenCalled();
        expect(spyNotify).toHaveBeenCalledWith('error', 'This is sample message');

        notificationDemo.find('#info-btn').simulate('click');
        expect(spyOnInfoMsg).toHaveBeenCalled();
        expect(spyNotify).toHaveBeenCalledWith('info', 'This is sample message');

        notificationDemo.find('input').simulate('change');
        expect(spyOnChangeMsgField).toHaveBeenCalled();
    });
});
