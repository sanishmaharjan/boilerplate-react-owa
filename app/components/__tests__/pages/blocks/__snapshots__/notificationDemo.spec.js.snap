// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationDemo renders correctly 1`] = `
<fieldset
  className="block-fieldset"
>
  <legend>
      
    Notification
      
  </legend>
  <p>
    <label>
      Message
      :
    </label>
    <input
      onChange={[Function]}
      type="text"
      value="This is sample message"
    />
  </p>
  <p>
    <button
      id="success-btn"
      onClick={[Function]}
    >
      Success
    </button>
    <button
      id="error-btn"
      onClick={[Function]}
    >
      Error
    </button>
    <button
      id="info-btn"
      onClick={[Function]}
    >
      Info
    </button>
  </p>
</fieldset>
`;

exports[`NotificationDemo renders correctly 2`] = `
<fieldset
  className="block-fieldset"
>
  <legend>
      
    Notification
      
  </legend>
  <p>
    <label>
      Message
      :
    </label>
    <input
      onChange={[Function]}
      type="text"
      value="New message"
    />
  </p>
  <p>
    <button
      id="success-btn"
      onClick={[Function]}
    >
      Success
    </button>
    <button
      id="error-btn"
      onClick={[Function]}
    >
      Error
    </button>
    <button
      id="info-btn"
      onClick={[Function]}
    >
      Info
    </button>
  </p>
</fieldset>
`;
